cmd/api/src/database/auth.go

```diff
@@ -534,12 +534,17 @@ func (s *BloodhoundDB) DeleteAuthSecret(ctx context.Context, authSecret model.Au
 // CreateUserSession creates a new UserSession row
 // INSERT INTO user_sessions (...) VALUES (..)
 func (s *BloodhoundDB) CreateUserSession(ctx context.Context, userSession model.UserSession) (model.UserSession, error) {
-	var (
-		newUserSession = userSession
-		result         = s.db.WithContext(ctx).Create(&newUserSession)
-	)
+	var newUserSession = userSession
+
+	err := s.db.WithContext(ctx).Transaction(func(tx *gorm.DB) error {
```

Comment:
Won't cause any harm, but do we need a transaction since this is a single DB call?